{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/maayanlab/Documents/Ovarian Cancer Project/ovarian-cancer\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import scipy as ss\n",
    "import pandas as pd\n",
    "import ipywidgets\n",
    "import matplotlib.pyplot as plt\n",
    "%cd /Users/maayanlab/Documents/Ovarian Cancer Project/ovarian-cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import widget classes and instatiate Network instance\n",
    "from clustergrammer_widget import *\n",
    "net = Network(clustergrammer_widget)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e78ca23b5d1b4905987d6ac47cbedaa1",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "clustergrammer_widget(network='{\"row_nodes\": [{\"name\": \"ADNP\", \"ini\": 300, \"clust\": 198, \"rank\": 140, \"rankvar…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load matrix file--for chea3 log transformed data\n",
    "net.load_file('./rawdata/df_tf_na_log3.txt')\n",
    "\n",
    "# cluster using default parameters\n",
    "net.cluster(enrichrgram=True)\n",
    "\n",
    "# net.cluster(views=[])\n",
    "\n",
    "# make interactive widget\n",
    "net.widget()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_test7 = pd.read_table(\"./rawdata/df_tf_na_log3.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "# fixing expression df for using with clustergrammer\n",
    "\n",
    "df_exp = pd.read_table('./rawdata/expression_forclustergrammer.txt').set_index(\"gene_symbol\")\n",
    "df_exp_na = df_exp.fillna(value = 0)\n",
    "df_exp_na.to_csv(\"./rawdata/expression_forclustergrammer2.txt\", sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-04-1331-01</th>\n",
       "      <th>TCGA-04-1332-01</th>\n",
       "      <th>TCGA-04-1338-01</th>\n",
       "      <th>TCGA-04-1341-01</th>\n",
       "      <th>TCGA-04-1343-01</th>\n",
       "      <th>TCGA-04-1347-01</th>\n",
       "      <th>TCGA-04-1350-01</th>\n",
       "      <th>TCGA-04-1356-01</th>\n",
       "      <th>TCGA-04-1357-01</th>\n",
       "      <th>TCGA-04-1361-01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-61-2101-01</th>\n",
       "      <th>TCGA-61-2102-01</th>\n",
       "      <th>TCGA-61-2104-01</th>\n",
       "      <th>TCGA-61-2109-01</th>\n",
       "      <th>TCGA-61-2110-01</th>\n",
       "      <th>TCGA-61-2111-01</th>\n",
       "      <th>TCGA-61-2113-01</th>\n",
       "      <th>TCGA-OY-A56Q-01</th>\n",
       "      <th>TCGA-VG-A8LO-01</th>\n",
       "      <th>TCGA-WR-A838-01</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MIR5702</th>\n",
       "      <td>-0.061008</td>\n",
       "      <td>-0.096135</td>\n",
       "      <td>-0.035748</td>\n",
       "      <td>-0.023189</td>\n",
       "      <td>-0.018332</td>\n",
       "      <td>-0.064984</td>\n",
       "      <td>-0.033921</td>\n",
       "      <td>-0.032554</td>\n",
       "      <td>-0.009093</td>\n",
       "      <td>-0.035971</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>-0.05017</td>\n",
       "      <td>0.003632</td>\n",
       "      <td>-0.040129</td>\n",
       "      <td>-0.046210</td>\n",
       "      <td>-0.029724</td>\n",
       "      <td>0.003889</td>\n",
       "      <td>-0.061887</td>\n",
       "      <td>-0.077884</td>\n",
       "      <td>-0.094632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCEB1P14</th>\n",
       "      <td>-0.061008</td>\n",
       "      <td>-0.096135</td>\n",
       "      <td>-0.035748</td>\n",
       "      <td>-0.023189</td>\n",
       "      <td>-0.018332</td>\n",
       "      <td>-0.064984</td>\n",
       "      <td>-0.033921</td>\n",
       "      <td>-0.032554</td>\n",
       "      <td>-0.009093</td>\n",
       "      <td>-0.035971</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>-0.05017</td>\n",
       "      <td>0.003632</td>\n",
       "      <td>-0.040129</td>\n",
       "      <td>-0.046210</td>\n",
       "      <td>-0.029724</td>\n",
       "      <td>0.003889</td>\n",
       "      <td>-0.061887</td>\n",
       "      <td>-0.077884</td>\n",
       "      <td>-0.094632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNA5SP27</th>\n",
       "      <td>-0.061008</td>\n",
       "      <td>-0.096135</td>\n",
       "      <td>-0.035748</td>\n",
       "      <td>-0.023189</td>\n",
       "      <td>-0.018332</td>\n",
       "      <td>-0.064984</td>\n",
       "      <td>-0.033921</td>\n",
       "      <td>-0.032554</td>\n",
       "      <td>-0.009093</td>\n",
       "      <td>-0.035971</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>-0.05017</td>\n",
       "      <td>0.003632</td>\n",
       "      <td>-0.040129</td>\n",
       "      <td>-0.046210</td>\n",
       "      <td>-0.029724</td>\n",
       "      <td>0.003889</td>\n",
       "      <td>-0.061887</td>\n",
       "      <td>-0.077884</td>\n",
       "      <td>-0.094632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDR42E1P1</th>\n",
       "      <td>-0.061008</td>\n",
       "      <td>-0.096135</td>\n",
       "      <td>-0.035748</td>\n",
       "      <td>-0.023189</td>\n",
       "      <td>-0.018332</td>\n",
       "      <td>-0.064984</td>\n",
       "      <td>-0.033921</td>\n",
       "      <td>-0.032554</td>\n",
       "      <td>-0.009093</td>\n",
       "      <td>-0.035971</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>-0.05017</td>\n",
       "      <td>0.003632</td>\n",
       "      <td>-0.040129</td>\n",
       "      <td>-0.046210</td>\n",
       "      <td>-0.029724</td>\n",
       "      <td>0.003889</td>\n",
       "      <td>-0.061887</td>\n",
       "      <td>-0.077884</td>\n",
       "      <td>-0.094632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCG</th>\n",
       "      <td>-0.071901</td>\n",
       "      <td>-0.104213</td>\n",
       "      <td>-0.043967</td>\n",
       "      <td>-0.031936</td>\n",
       "      <td>-0.028854</td>\n",
       "      <td>-0.072680</td>\n",
       "      <td>-0.042582</td>\n",
       "      <td>-0.043065</td>\n",
       "      <td>-0.018861</td>\n",
       "      <td>-0.047670</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003503</td>\n",
       "      <td>-0.05724</td>\n",
       "      <td>-0.008308</td>\n",
       "      <td>-0.051223</td>\n",
       "      <td>-0.055039</td>\n",
       "      <td>-0.038468</td>\n",
       "      <td>-0.007739</td>\n",
       "      <td>-0.070373</td>\n",
       "      <td>-0.088028</td>\n",
       "      <td>-0.103011</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 379 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             TCGA-04-1331-01  TCGA-04-1332-01  TCGA-04-1338-01  \\\n",
       "gene_symbol                                                      \n",
       "MIR5702            -0.061008        -0.096135        -0.035748   \n",
       "TCEB1P14           -0.061008        -0.096135        -0.035748   \n",
       "RNA5SP27           -0.061008        -0.096135        -0.035748   \n",
       "SDR42E1P1          -0.061008        -0.096135        -0.035748   \n",
       "GCG                -0.071901        -0.104213        -0.043967   \n",
       "\n",
       "             TCGA-04-1341-01  TCGA-04-1343-01  TCGA-04-1347-01  \\\n",
       "gene_symbol                                                      \n",
       "MIR5702            -0.023189        -0.018332        -0.064984   \n",
       "TCEB1P14           -0.023189        -0.018332        -0.064984   \n",
       "RNA5SP27           -0.023189        -0.018332        -0.064984   \n",
       "SDR42E1P1          -0.023189        -0.018332        -0.064984   \n",
       "GCG                -0.031936        -0.028854        -0.072680   \n",
       "\n",
       "             TCGA-04-1350-01  TCGA-04-1356-01  TCGA-04-1357-01  \\\n",
       "gene_symbol                                                      \n",
       "MIR5702            -0.033921        -0.032554        -0.009093   \n",
       "TCEB1P14           -0.033921        -0.032554        -0.009093   \n",
       "RNA5SP27           -0.033921        -0.032554        -0.009093   \n",
       "SDR42E1P1          -0.033921        -0.032554        -0.009093   \n",
       "GCG                -0.042582        -0.043065        -0.018861   \n",
       "\n",
       "             TCGA-04-1361-01       ...         TCGA-61-2101-01  \\\n",
       "gene_symbol                        ...                           \n",
       "MIR5702            -0.035971       ...                0.012824   \n",
       "TCEB1P14           -0.035971       ...                0.012824   \n",
       "RNA5SP27           -0.035971       ...                0.012824   \n",
       "SDR42E1P1          -0.035971       ...                0.012824   \n",
       "GCG                -0.047670       ...                0.003503   \n",
       "\n",
       "             TCGA-61-2102-01  TCGA-61-2104-01  TCGA-61-2109-01  \\\n",
       "gene_symbol                                                      \n",
       "MIR5702             -0.05017         0.003632        -0.040129   \n",
       "TCEB1P14            -0.05017         0.003632        -0.040129   \n",
       "RNA5SP27            -0.05017         0.003632        -0.040129   \n",
       "SDR42E1P1           -0.05017         0.003632        -0.040129   \n",
       "GCG                 -0.05724        -0.008308        -0.051223   \n",
       "\n",
       "             TCGA-61-2110-01  TCGA-61-2111-01  TCGA-61-2113-01  \\\n",
       "gene_symbol                                                      \n",
       "MIR5702            -0.046210        -0.029724         0.003889   \n",
       "TCEB1P14           -0.046210        -0.029724         0.003889   \n",
       "RNA5SP27           -0.046210        -0.029724         0.003889   \n",
       "SDR42E1P1          -0.046210        -0.029724         0.003889   \n",
       "GCG                -0.055039        -0.038468        -0.007739   \n",
       "\n",
       "             TCGA-OY-A56Q-01  TCGA-VG-A8LO-01  TCGA-WR-A838-01  \n",
       "gene_symbol                                                     \n",
       "MIR5702            -0.061887        -0.077884        -0.094632  \n",
       "TCEB1P14           -0.061887        -0.077884        -0.094632  \n",
       "RNA5SP27           -0.061887        -0.077884        -0.094632  \n",
       "SDR42E1P1          -0.061887        -0.077884        -0.094632  \n",
       "GCG                -0.070373        -0.088028        -0.103011  \n",
       "\n",
       "[5 rows x 379 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_exp_na.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a6eaac1d86814aa3bb5577db57603023",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "clustergrammer_widget(network='{\"row_nodes\": [{\"name\": \"MIR5702\", \"ini\": 1000, \"clust\": 335, \"rank\": 997, \"ran…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load matrix file--for expression log transformed data--check this data--took top 1000 variant genes but not sure if variance was calculated correctly\n",
    "net.load_file('./rawdata/expression_forclustergrammer2.txt')\n",
    "\n",
    "# cluster using default parameters\n",
    "net.cluster()\n",
    "\n",
    "# make interactive widget\n",
    "net.widget()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/maayanlab/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/calc_clust.py:60: RuntimeWarning: invalid value encountered in less\n",
      "  inst_dm[inst_dm < 0] = float(0)\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "The condensed distance matrix must contain only finite values.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-61-21aedf871865>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m# cluster using default parameters\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mnet\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcluster\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;31m# make interactive widget\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/__init__.py\u001b[0m in \u001b[0;36mcluster\u001b[0;34m(self, dist_type, run_clustering, dendro, views, linkage_type, sim_mat, filter_sim, calc_cat_pval, run_enrichr, enrichrgram)\u001b[0m\n\u001b[1;32m     97\u001b[0m                                    \u001b[0mcalc_cat_pval\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcalc_cat_pval\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     98\u001b[0m                                    \u001b[0mrun_enrichr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrun_enrichr\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 99\u001b[0;31m                                    enrichrgram=enrichrgram)\n\u001b[0m\u001b[1;32m    100\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    101\u001b[0m   def make_clust(self, dist_type='cosine', run_clustering=True,\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/make_clust_fun.py\u001b[0m in \u001b[0;36mmake_clust\u001b[0;34m(net, dist_type, run_clustering, dendro, requested_views, linkage_type, sim_mat, filter_sim, calc_cat_pval, sim_mat_views, run_enrichr, enrichrgram)\u001b[0m\n\u001b[1;32m     36\u001b[0m                                 \u001b[0mrun_clustering\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrun_clustering\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     37\u001b[0m                                 \u001b[0mdendro\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdendro\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mignore_cat\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 38\u001b[0;31m                                 calc_cat_pval=calc_cat_pval)\n\u001b[0m\u001b[1;32m     39\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     40\u001b[0m   \u001b[0mall_views\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/calc_clust.py\u001b[0m in \u001b[0;36mcluster_row_and_col\u001b[0;34m(net, dist_type, linkage_type, dendro, run_clustering, run_rank, ignore_cat, calc_cat_pval, links)\u001b[0m\n\u001b[1;32m     24\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mrun_clustering\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m       \u001b[0mnode_info\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'clust'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnode_info\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'group'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m\\\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 26\u001b[0;31m           \u001b[0mclust_and_group\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnet\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdm\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0minst_rc\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlinkage_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlinkage_type\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     27\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m       \u001b[0mdendro\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/calc_clust.py\u001b[0m in \u001b[0;36mclust_and_group\u001b[0;34m(net, inst_dm, linkage_type)\u001b[0m\n\u001b[1;32m     64\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mclust_and_group\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnet\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minst_dm\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlinkage_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'average'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m   \u001b[0;32mimport\u001b[0m \u001b[0mscipy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcluster\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhierarchy\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mhier\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 66\u001b[0;31m   \u001b[0mY\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhier\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlinkage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minst_dm\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlinkage_type\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     67\u001b[0m   \u001b[0mZ\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhier\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdendrogram\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mY\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mno_plot\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     68\u001b[0m   \u001b[0minst_clust_order\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mZ\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'leaves'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/scipy/cluster/hierarchy.py\u001b[0m in \u001b[0;36mlinkage\u001b[0;34m(y, method, metric, optimal_ordering)\u001b[0m\n\u001b[1;32m    711\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    712\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misfinite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 713\u001b[0;31m         raise ValueError(\"The condensed distance matrix must contain only \"\n\u001b[0m\u001b[1;32m    714\u001b[0m                          \"finite values.\")\n\u001b[1;32m    715\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: The condensed distance matrix must contain only finite values."
     ]
    }
   ],
   "source": [
    "# load matrix file--for expression log transformed data--check this data--took top 1000 variant genes but not sure if variance was calculated correctly\n",
    "net.load_file('./rawdata/expression_forclustergrammer.txt')\n",
    "\n",
    "# cluster using default parameters\n",
    "net.cluster()\n",
    "\n",
    "# make interactive widget\n",
    "net.widget()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_test5 = pd.read_table(\"./rawdata/expression_forclustergrammer.txt\").set_index(\"gene_symbol\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_test5.to_csv(\"./rawdata/expression_forclustergrammer.txt\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on Network in module clustergrammer_widget.clustergrammer object:\n",
      "\n",
      "class Network(builtins.object)\n",
      " |  version 1.13.5\n",
      " |  \n",
      " |  Clustergrammer.py takes a matrix as input (either from a file of a Pandas DataFrame), normalizes/filters, hierarchically clusters, and produces the :ref:`visualization_json` for :ref:`clustergrammer_js`.\n",
      " |  \n",
      " |  Networks have two states:\n",
      " |  \n",
      " |    1. the data state, where they are stored as a matrix and nodes\n",
      " |    2. the viz state where they are stored as viz.links, viz.row_nodes, and viz.col_nodes.\n",
      " |  \n",
      " |  The goal is to start in a data-state and produce a viz-state of\n",
      " |  the network that will be used as input to clustergram.js.\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  Iframe_web_app(self, filename=None, width=1000, height=800)\n",
      " |  \n",
      " |  __init__(self, widget=None)\n",
      " |      Initialize self.  See help(type(self)) for accurate signature.\n",
      " |  \n",
      " |  add_cats(self, axis, cat_data)\n",
      " |      Add categories to rows or columns using cat_data array of objects. Each object in cat_data is a dictionary with one key (category title) and value (rows/column names) that have this category. Categories will be added onto the existing categories and will be added in the order of the objects in the array.\n",
      " |      \n",
      " |      Example ``cat_data``::\n",
      " |      \n",
      " |      \n",
      " |          [\n",
      " |            {\n",
      " |              \"title\": \"First Category\",\n",
      " |              \"cats\": {\n",
      " |                \"true\": [\n",
      " |                  \"ROS1\",\n",
      " |                  \"AAK1\"\n",
      " |                ]\n",
      " |              }\n",
      " |            },\n",
      " |            {\n",
      " |              \"title\": \"Second Category\",\n",
      " |              \"cats\": {\n",
      " |                \"something\": [\n",
      " |                  \"PDK4\"\n",
      " |                ]\n",
      " |              }\n",
      " |            }\n",
      " |          ]\n",
      " |  \n",
      " |  clip(self, lower=None, upper=None)\n",
      " |      Trim values at input thresholds using pandas function\n",
      " |  \n",
      " |  cluster(self, dist_type='cosine', run_clustering=True, dendro=True, views=['N_row_sum', 'N_row_var'], linkage_type='average', sim_mat=False, filter_sim=0.1, calc_cat_pval=False, run_enrichr=None, enrichrgram=None)\n",
      " |      The main function performs hierarchical clustering, optionally generates filtered views (e.g. row-filtered views), and generates the :``visualization_json``.\n",
      " |  \n",
      " |  dat_to_df(self)\n",
      " |      Export Pandas DataFrams (will be deprecated).\n",
      " |  \n",
      " |  dendro_cats(self, axis, dendro_level)\n",
      " |      Generate categories from dendrogram groups/clusters. The dendrogram has 11\n",
      " |      levels to choose from 0 -> 10. Dendro_level can be given as an integer or\n",
      " |      string.\n",
      " |  \n",
      " |  df_to_dat(self, df, define_cat_colors=False)\n",
      " |      Load Pandas DataFrame (will be deprecated).\n",
      " |  \n",
      " |  downsample(self, df=None, ds_type='kmeans', axis='row', num_samples=100, random_state=1000)\n",
      " |      Downsample the matrix rows or columns (currently supporting kmeans only). Users can optionally pass in a DataFrame to be downsampled (and this will be incorporated into the network object).\n",
      " |  \n",
      " |  enrichrgram(self, lib, axis='row')\n",
      " |      Add Enrichr gene enrichment results to your visualization (where your rows\n",
      " |      are genes). Run enrichrgram before clustering to incldue enrichment results\n",
      " |      as row categories. Enrichrgram can also be run on the front-end using the\n",
      " |      Enrichr logo at the top left.\n",
      " |      \n",
      " |      Set lib to the Enrichr library that you want to use for enrichment analysis.\n",
      " |      Libraries included:\n",
      " |      \n",
      " |        * ChEA_2016\n",
      " |        * KEA_2015\n",
      " |        * ENCODE_TF_ChIP-seq_2015\n",
      " |        * ENCODE_Histone_Modifications_2015\n",
      " |        * Disease_Perturbations_from_GEO_up\n",
      " |        * Disease_Perturbations_from_GEO_down\n",
      " |        * GO_Molecular_Function_2015\n",
      " |        * GO_Biological_Process_2015\n",
      " |        * GO_Cellular_Component_2015\n",
      " |        * Reactome_2016\n",
      " |        * KEGG_2016\n",
      " |        * MGI_Mammalian_Phenotype_Level_4\n",
      " |        * LINCS_L1000_Chem_Pert_up\n",
      " |        * LINCS_L1000_Chem_Pert_down\n",
      " |  \n",
      " |  export_df(self)\n",
      " |      Export Pandas DataFrame/\n",
      " |  \n",
      " |  export_net_json(self, net_type='viz', indent='no-indent')\n",
      " |      Export dat or viz JSON.\n",
      " |  \n",
      " |  export_viz_to_widget(self, which_viz='viz')\n",
      " |      Export viz JSON, for use with clustergrammer_widget. Formerly method was\n",
      " |      named widget.\n",
      " |  \n",
      " |  filter_N_top(self, inst_rc, N_top, rank_type='sum')\n",
      " |      Filter the matrix rows or columns based on sum/variance, and only keep the top\n",
      " |      N.\n",
      " |  \n",
      " |  filter_cat(self, axis, cat_index, cat_name)\n",
      " |      Filter the matrix based on their category. cat_index is the index of the category, the first category has index=1.\n",
      " |  \n",
      " |  filter_names(self, axis, names)\n",
      " |      Filter the visualization using row/column names. The function takes, axis ('row'/'col') and names, a list of strings.\n",
      " |  \n",
      " |  filter_sum(self, inst_rc, threshold, take_abs=True)\n",
      " |      Filter a network's rows or columns based on the sum across rows or columns.\n",
      " |  \n",
      " |  filter_threshold(self, inst_rc, threshold, num_occur=1)\n",
      " |      Filter the matrix rows or columns based on num_occur values being above a\n",
      " |      threshold (in absolute value).\n",
      " |  \n",
      " |  load_data_file_to_net(self, filename)\n",
      " |      Load Clustergrammer's dat format (saved as JSON).\n",
      " |  \n",
      " |  load_df(self, df)\n",
      " |      Load Pandas DataFrame.\n",
      " |  \n",
      " |  load_file(self, filename)\n",
      " |      Load TSV file.\n",
      " |  \n",
      " |  load_file_as_string(self, file_string, filename='')\n",
      " |      Load file as a string.\n",
      " |  \n",
      " |  load_stdin(self)\n",
      " |      Load stdin TSV-formatted string.\n",
      " |  \n",
      " |  load_tsv_to_net(self, file_buffer, filename=None)\n",
      " |      This will load a TSV matrix file buffer; this is exposed so that it will\n",
      " |      be possible to load data without having to read from a file.\n",
      " |  \n",
      " |  load_vect_post_to_net(self, vect_post)\n",
      " |      Load data in the vector format JSON.\n",
      " |  \n",
      " |  make_clust(self, dist_type='cosine', run_clustering=True, dendro=True, views=['N_row_sum', 'N_row_var'], linkage_type='average', sim_mat=False, filter_sim=0.1, calc_cat_pval=False, run_enrichr=None, enrichrgram=None)\n",
      " |      ... Will be deprecated, renaming method cluster ...\n",
      " |      The main function performs hierarchical clustering, optionally generates filtered views (e.g. row-filtered views), and generates the :``visualization_json``.\n",
      " |  \n",
      " |  normalize(self, df=None, norm_type='zscore', axis='row', keep_orig=False)\n",
      " |      Normalize the matrix rows or columns using Z-score (zscore) or Quantile Normalization (qn). Users can optionally pass in a DataFrame to be normalized (and this will be incorporated into the Network object).\n",
      " |  \n",
      " |  produce_view(self, requested_view=None)\n",
      " |      This function is under development and will produce a single view on demand.\n",
      " |  \n",
      " |  random_sample(self, num_samples, df=None, replace=False, weights=None, random_state=100, axis='row')\n",
      " |      Return random sample of matrix.\n",
      " |  \n",
      " |  reset(self)\n",
      " |      This re-initializes the Network object.\n",
      " |  \n",
      " |  set_cat_color(self, axis, cat_index, cat_name, inst_color)\n",
      " |  \n",
      " |  swap_nan_for_zero(self)\n",
      " |      Swaps all NaN (numpy NaN) instances for zero.\n",
      " |  \n",
      " |  widget(self, which_viz='viz')\n",
      " |      Generate a widget visualization using the widget. The export_viz_to_widget\n",
      " |      method passes the visualization JSON to the instantiated widget, which is\n",
      " |      returned and visualized on the front-end.\n",
      " |  \n",
      " |  widget_df(self)\n",
      " |      Export a DataFrame from the front-end visualization. For instance, a user\n",
      " |      can filter to show only a single cluster using the dendrogram and then\n",
      " |      get a dataframe of this cluster using the widget_df method.\n",
      " |  \n",
      " |  write_json_to_file(self, net_type, filename, indent='no-indent')\n",
      " |      Save dat or viz as a JSON to file.\n",
      " |  \n",
      " |  write_matrix_to_tsv(self, filename=None, df=None)\n",
      " |      Export data-matrix to file.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Static methods defined here:\n",
      " |  \n",
      " |  load_gmt(filename)\n",
      " |  \n",
      " |  load_json_to_dict(filename)\n",
      " |  \n",
      " |  save_dict_to_json(inst_dict, filename, indent='no-indent')\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(net)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "from clustergrammer import run\n",
    "from clustergrammer import plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'logCPM'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3062\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3063\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3064\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'logCPM'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-47-f3cdc883480a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mdf_tf_na\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_table\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"./rawdata/df_tf_na_log3.txt\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mclustergrammer_results\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrun\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_tf_na\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclustergrammer_results\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/Ovarian Cancer Project/ovarian-cancer/clustergrammer.py\u001b[0m in \u001b[0;36mrun\u001b[0;34m(dataset, normalization, z_score, nr_genes, metadata_cols)\u001b[0m\n\u001b[1;32m     32\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m         \u001b[0;31m# Get data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 34\u001b[0;31m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdataset\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     35\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     36\u001b[0m         \u001b[0;31m# Get tempfile\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2683\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2684\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2685\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2686\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2687\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_getitem_column\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2690\u001b[0m         \u001b[0;31m# get column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2691\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2692\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2693\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2694\u001b[0m         \u001b[0;31m# duplicate columns & possible reduce dimensionality\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_get_item_cache\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   2484\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcache\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2485\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2486\u001b[0;31m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2487\u001b[0m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_box_item_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2488\u001b[0m             \u001b[0mcache\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/internals.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, item, fastpath)\u001b[0m\n\u001b[1;32m   4113\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4114\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4115\u001b[0;31m                 \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4116\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4117\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3063\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3064\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3065\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_cast_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3066\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3067\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtolerance\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtolerance\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'logCPM'"
     ]
    }
   ],
   "source": [
    "df_tf_na = pd.read_table(\"./rawdata/df_tf_na_log3.txt\")\n",
    "clustergrammer_results = run(df_tf_na)\n",
    "plot(clustergrammer_results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "bad operand type for abs(): 'str'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-52-271c51a0a6c3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mnet\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_tf_na\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mnet\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcluster\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0menrichrgram\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mnet\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwidget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/__init__.py\u001b[0m in \u001b[0;36mcluster\u001b[0;34m(self, dist_type, run_clustering, dendro, views, linkage_type, sim_mat, filter_sim, calc_cat_pval, run_enrichr, enrichrgram)\u001b[0m\n\u001b[1;32m     97\u001b[0m                                    \u001b[0mcalc_cat_pval\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcalc_cat_pval\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     98\u001b[0m                                    \u001b[0mrun_enrichr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrun_enrichr\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 99\u001b[0;31m                                    enrichrgram=enrichrgram)\n\u001b[0m\u001b[1;32m    100\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    101\u001b[0m   def make_clust(self, dist_type='cosine', run_clustering=True,\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/make_clust_fun.py\u001b[0m in \u001b[0;36mmake_clust\u001b[0;34m(net, dist_type, run_clustering, dendro, requested_views, linkage_type, sim_mat, filter_sim, calc_cat_pval, sim_mat_views, run_enrichr, enrichrgram)\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m   \u001b[0mthreshold\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0.0001\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 19\u001b[0;31m   \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrun_filter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdf_filter_row_sum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mthreshold\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     20\u001b[0m   \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrun_filter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdf_filter_col_sum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mthreshold\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/clustergrammer_widget/clustergrammer/run_filter.py\u001b[0m in \u001b[0;36mdf_filter_row_sum\u001b[0;34m(df, threshold, take_abs)\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m   \u001b[0;32mif\u001b[0m \u001b[0mtake_abs\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0mdf_copy\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdeepcopy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'mat'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mabs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m   \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m     \u001b[0mdf_copy\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdeepcopy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'mat'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36mabs\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   8256\u001b[0m         \u001b[0mnumpy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mabsolute\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mcalculate\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mabsolute\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0melement\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mwise\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   8257\u001b[0m         \"\"\"\n\u001b[0;32m-> 8258\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mabs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   8259\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   8260\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mdescribe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpercentiles\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minclude\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexclude\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: bad operand type for abs(): 'str'"
     ]
    }
   ],
   "source": [
    "net.load_df(df_tf_na)\n",
    "net.cluster(enrichrgram=True)\n",
    "net.widget()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
